### PR TITLE
add ssl support for graph daemon

### PR DESCRIPTION
**Pull request by klay-ke**

### CODE REVIEW

File: client/src/main/java/com/vesoft/nebula/client/graph/net/SyncConnection.java [hunk 1]
```diff
@@ -192,7 +188,6 @@ public boolean ping() {
             execute(0, "YIELD 1;");
             return true;
         } catch (IOErrorException e) {
-            e.printStackTrace();
             return false;
```
Comment (Nicole00) [inline client/src/main/java/com/vesoft/nebula/client/graph/net/SyncConnection.java]: please add some log print When ping() returns false， log will help to locate the problem.

File: client/src/main/java/com/vesoft/nebula/util/SslUtil.java [hunk 2]
```diff
@@ -100,7 +103,7 @@ public static SSLSocketFactory getSSLSocketFactoryWithCA(CASignedSSLParam param)
             return context.getSocketFactory();
 
         } catch (Exception e) {
-            e.printStackTrace();
+            LOGGER.error(e.getMessage());
```
Comment (Nicole00) [inline client/src/main/java/com/vesoft/nebula/util/SslUtil.java:106]: it's better to close in `finally{}`.  In your code, if reader.readObject throws some exception, then the reader cannot be closed.

**Comment by darionyaphet** [comment]
I recommend submitting thrift generated code separately

File: client/src/test/resources/docker-compose-selfsigned.yaml [hunk 3]
```diff
@@ -0,0 +1,135 @@
+version: '3.4'
+services:
+  metad-selfsigned:
+    image: vesoft/nebula-metad:nightly
+    environment:
+      USER: root
+      TZ:   "${TZ}"
+    command:
+      - --meta_server_addrs=172.29.1.1:8559
+      - --local_ip=172.29.1.1
```
[the remark this answers is not in the document]
Comment (klay-ke) [reply client/src/test/resources/docker-compose-selfsigned.yaml:10]: Again, I don't get the point, I've defined networks:
    networks:
      nebula-net-selfsigned:
        ipv4_address: 172.29.1.1

what does localhost mean here?

**Pull request merged**